library/std/src/process.rs

```diff
@@ -2294,6 +2294,20 @@ impl Child {
 /// considered undesirable. Note that returning from `main` also calls `exit`, so making `exit` an
 /// unsafe operation is not an option.)
 ///
+/// ## Safe interop with C code
+///
+/// This function is safe to call as long as `exit` is only ever invoked from Rust. However, the C
```

Comment:
If you have two rust cdylibs which concurrently call exit you did get the same issue, right? They don't share the exit lock but rather have copies as the standard library is copied too.

Reply:
Hm, yeah... if we list all corner cases the docs will explode :/  It's so much nicer when things are just compositional.^^